airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigPersistenceFactory.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.config.persistence;
+
+import static io.airbyte.config.persistence.AirbyteConfigsTable.AIRBYTE_CONFIGS_TABLE_SCHEMA;
+
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.config.Configs;
+import io.airbyte.db.Database;
+import io.airbyte.db.Databases;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * By default, this factory returns a database config persistence. it can still return a file system
+ * config persistence for testing purpose. This legacy feature should be removed after the file to
+ * database migration is completely done.
```

Comment:
I'm thinking it would be really nice to keep a working file-based config persistence and not necessarily remove it permanently...

If this file config persistence reads/writes from a docker volume that is mounted from a local folder of the filesystem (like the `LOCAL_ROOT` does) rather than a named docker volume, then a user could configure a CI running as part of their git repository versioning their airbyte setup. This setup could include (tests) configurations and once the CI is successful, they could publish in the same workflow their final changes to a remote Airbyte Database config persistence that would affect their airbyte instance in production for example.

WDYT?

Reply:
Just curious, what benefits do you see from working with a file-based config persistence (assuming we have import/export capabilities)?

Reply:
I guess what Chris means is that if we keep the file config persistence code, users can still manipulate the configs using the code.

Reply:
For import/export capabilities, we'd have what is currently exposed as: 
- `getDbPersistenceWithYamlSeed`: (standard files released with airbyte)
- `getDbPersistenceWithFileSeed`: (old named docker volumes)

In the future, we could have something like these?
- `getDbPersistenceWithYamlSeedFromGitRepo`: clones a remote git repo where seeds are stored to re-init the DB from those
- `getDbPersistenceWithYamlSeedFromArchive`: same but from a local/remote .tar.gz archive file
- `getDbPersistenceWithYamlSeedFromLocalFolder`: same but from a local folder (no need to compress into an archive or push to a repo?)

> Just curious, what benefits do you see from working with a file-based config persistence vs just using import/export capabilities?
> if we keep the file config persistence code, users can still manipulate the configs using the code?

Yes.

For dev/test purposes, you can more easily edit/test your airbyte configurations locally by manually/programmatically editing the configs in parallel without having to restart airbyte all the time to re-init the database (re-deploy your changes)? 

Plus the code is almost already written for that?
Of course, for production usage, it's better to go through the database